src/core/CMakeLists.txt

```diff
@@ -101,6 +123,13 @@ set(libraryname idyntree-core)
 
 add_library(${libraryname} ${IDYNTREE_CORE_EXP_SOURCES} ${IDYNTREE_CORE_EXP_HEADERS} ${IDYNTREE_CORE_EXP_PRIVATE_INCLUDES})
 
+if (DEFINED ${CMAKE_COMPILER_IS_GNUCXX} 
```

Comment:
DEFINED takes the variable, not the variable value, so I think this needs to be `DEFINED CMAKE_COMPILER_IS_GNUCXX`